### ISSUE TITLE
review: inactive prefs

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#138]
There are currently 86 matches for "// user_" (up to 2800 inclusive) - we should check them all for consistency - eg why? are tags applied [warning], [setup] etc? are [notes] included? etc. Are we enforcing defaults consistently?

### Consider (39 items in 32 pref numbers)

:small_orange_diamond: pref value differs from the default but we give no warning/note (or even a setup) as to why we left it inactive
 * [x] 1003: disable memory cache
   // user_pref("browser.cache.memory.enable", false);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/1f18162d562433d0cc86090af5bcc8e111179d99
 * [x] 1005: disable fastback cache
   // user_pref("browser.sessionhistory.max_total_viewers", 0);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/1f18162d562433d0cc86090af5bcc8e111179d99
 * [x] 1006: disable permissions manager from writing to disk (requires restart)
   // user_pref("permissions.memory_only", true);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/1f18162d562433d0cc86090af5bcc8e111179d99
 * [ ] 1031: disable favicons in tabs and new bookmarks
   // user_pref("browser.chrome.site_icons", false);
   // user_pref("browser.chrome.favicons", false);

:small_blue_diamond:  pref value is the same as default, why don't we just enforce it
 * [ ] 1008: set DNS cache and expiration time (default 400 and 60, same as TBB)
   // user_pref("network.dnsCacheEntries", 400);
   // user_pref("network.dnsCacheExpiration", 60);
 * [x] 1032: disable favicons in web notifications
   // user_pref("alerts.showFavicons", false);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/eeedf0db729c1bd5196709ab0044ca5881b6d9f4 - enforced default
 * [x] 1409: limit system font exposure to a whitelist (FF52+) [SETUP]
   // user_pref("font.system.whitelist", "");
   * `Thorin says`:  not used because no-one has a consensus of what to put in, has a [note], may be covered by resistFingerprinting, and user.js limits fonts anyway in 1401. We can leave this as is.
 * [x] 1606: ALL: set the default Referrer Policy (FF53+)
   // user_pref("network.http.referer.userControlPolicy", 3);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/1ad970741caac3f18c08e1a4c98e45208f3a6611 - enforced default
 * [ ] 2450c: set size of warning quota for offline cache (default 51200)
   // user_pref("offline-apps.quota.warn", 51200);
 * [ ] 2703: set cookie lifetime policy
   // user_pref("network.cookie.lifetimePolicy", 0);
 * [ ] 2704: set cookie lifetime in days (see above pref) - default is 90 days
   // user_pref("network.cookie.lifetime.days", 90);

---
 * [x] 0102: set start page
   // user_pref("browser.startup.page", 0);
   * FYI only - not messing with people's start pages
 * [x] 0850a: disable location bar autocomplete [controlled by 0850b]
   // user_pref("browser.urlbar.autocomplete.enabled", false);
   * controlled by other prefs - differs from default
 * [x] 0850c: disable location bar dropdown
   // user_pref("browser.urlbar.maxRichResults", 0);
   * has note - differs from default
 * [x] 0860: disable search and form history
   // user_pref("browser.formfill.enable", false);
   * differs from default, user choice, section header has setup tag
 * [x] 0861: disable saving form history on secure websites
   // user_pref("browser.formfill.saveHttpsForms", false);
   * has info on why to leave alone at true - differs from default
 * [x] 0862: disable browsing and download history
   // user_pref("places.history.enabled", false);
   * differs from default, user choice, section header has setup tag
 * [x] 0901: disable saving passwords
   // user_pref("signon.rememberSignons", false);
   * differs from default, user choice, section header has setup tag
 * [x] 1201: disable old SSL/TLS - vulnerable to a MiTM attack
   // user_pref("security.ssl.require_safe_negotiation", true);
   * has warning, breakage - differs from default
 * [x] 1202: control TLS versions with min and max
   // user_pref("security.tls.version.min", 2);
   // user_pref("security.tls.version.fallback-limit", 3);
   // user_pref("security.tls.version.max", 4); // 4 = allow up to and including TLS 1.3
   * min has a warning - differs from default
   * fallback and max now enforced see https://github.com/ghacksuserjs/ghacks-user.js/commit/0006ee04b4bc3470894059a85727ef7f513548e0
 * [x] 1212: enable OCSP revocation
   // user_pref("security.OCSP.require", true);
   * has warning - differs from default
 * [x] 1221: disable intermediate certificate caching (fingerprinting attack vector)
   // user_pref("security.nocertdb", true);
   * has warning - hidden pref - definitely changes behavior browser so not default
 * [x] 1241: disable insecure passive content (such as images) on https pages - mixed context
   // user_pref("security.mixed_content.block_display_content", true);
   * has warning, breakage - differs from default
 * [x] 1264: disable the remaining non-modern cipher suites as of FF52
   // user_pref("security.ssl3.rsa_aes_128_sha", false);
   // user_pref("security.ssl3.rsa_aes_256_sha", false);
   * has a note - differ from defaults
 * [x] 1803: set a plugin state: 0=deactivated 1=ask 2=enabled (Flash example)
   // user_pref("plugin.state.flash", 0);
   * example only, we recommend no flash - differs from default
 * [x] 2201: disable website control over browser right-click context menu
   // user_pref("dom.event.contextmenu.enabled", false);
   * has note, breakage (is almost useless as a pref now) - differs from default
 * [x] 2421: disable Ion and baseline JIT
   // user_pref("javascript.options.ion", false);
   // user_pref("javascript.options.baselinejit", false);
   * has warning, breakage - differs from default
 * [x] 2514: spoof (or limit?) number of CPU cores (also see 2699f)
   // user_pref("dom.maxHardwareConcurrency", 2);
   * has warning, not recommended - differs from default depending on PC
 * [x] 2621: disable IPv6 (included for knowledge ONLY [WARNING] do not do this)
   // user_pref("network.dns.disableIPv6", true);
   // user_pref("network.http.fast-fallback-to-IPv4", true);
   * has warning, not recommended - differs from default
 * [x] 2671: disable in-content SVG (Scalable Vector Graphics) (FF53+)
   // user_pref("svg.disabled", true);
   * has warning, breakage - differs from default
 * [x] 2705: disable dom storage
   // user_pref("dom.storage.enabled", false);
   * has warning, breakage - differs from default
 * [x] 2804: set what history items to clear with Ctrl-Shift-Del
   // user_pref("privacy.cpd.downloads", true); // not used, see note above
   * not used, controlled by another pref

### While we're at it

Would like these changed from default false to inactive true, fix the description (in the last week of bugzilla reading I can confirm without any refs, that openWIndows refers to sessiondata/session restore), and add in that there is a long standing issue unlikely to be resolved re double windows on launch (maybe it's ok on linux or something, I am still amazed why it's enforced in PK's as true - already three issues raised over it, no doubt many more to come. We also have session restore disabled anyway)

* [x]  2805: privacy.*.openWindows (FF34+)
   // user_pref("privacy.clearOnShutdown.openWindows", false);
   // user_pref("privacy.cpd.openWindows", false);
   * `DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/b9e321c45aff9456dd53b65145b2aa07d854b55c

### Ignore (47 items in 29 pref numbers)

PB start mode, auto-update checks, Kinto+SB+TB etc, e10s, containers, UA spoofing, FPI, and user choice on window res (almost all of these differ from default)

 * 0001: start Firefox in PB (Private Browsing) mode
 * 0301a: disable auto-update checks for Firefox
 * 0301b: disable auto-update checks for add-ons
 * 0403: disable individual Kinto blocklists -  // 4 prefs
 * 0410: disable "Block dangerous and deceptive content" -  // 2 prefs
 * 0411: disable "Block dangerous downloads"
 * 0412: disable "Warn me about unwanted and uncommon software" - // 4 prefs
 * 0413: disable Google safebrowsing updates - // 4 prefs
 * 0416: disable 'ignore this warning' on Safe Browsing warnings which when clicked
 * 0422: enable "basic" or "strict" tracking protecting list - // 2 prefs
 * 0423: disable Mozilla's blocklist for known Flash tracking/fingerprinting
 * 0424: disable Mozilla's tracking protection and Flash blocklist updates - // 2 prefs
 * 1101: start the browser in e10s mode (FF48+) - // 4 prefs
 * 1102: control number of e10s processes
 * 1103: enable WebExtension add-on code to run in a separate process
 * 1104: enforce separate content process for file://URLs
 * 1110: set sandbox level. DO NOT MEDDLE WITH THESE - // 3 prefs
 * 1701: enable [SETTING] Options>Privacy>Container Tabs
 * 1702: enable Container Tabs
 * 1703: enable a private container for thumbnail loads
 * 1704: set long press behaviour on "+ Tab" button to display container menu
 * 2697a: navigator.userAgent
 * 2697b: navigator.buildID
 * 2697c: navigator.appName
 * 2697d: navigator.appVersion
 * 2697e: navigator.platform
 * 2697f: navigator.oscpu
 * 2698*: FPI - // 2 prefs
 * 2699d: set new window sizes to round to hundreds - // 2 prefs user choice

...

**Comment by Thorin-Oakenpants** [comment]
PS: This is a good list to see if you have these 86 prefs reset in about:config. I did the same thing with the deprecated sticky and found one I had missed about a year ago (as well as some old `plugin.state*` items)

**Comment by Thorin-Oakenpants** [comment]
:+1: or nits - 2805 rewrite

```js
/* 2805: privacy.*.openWindows (clear session restore data) (FF34+)
 * [WARNING] There is a long-standing years-old bug that these cause two windows on Firefox
 * restart. You do not need these anyway if session restore is diabled (see 1020) ***/
   // user_pref("privacy.clearOnShutdown.openWindows", true);
   // user_pref("privacy.cpd.openWindows", true);
```

**Comment by earthlng** [comment]
* 1005: disable fastback cache - inactive because of usability and convenience
* 1006: disable permissions manager from writing to disk (requires restart)
inactive because this templates with its default values relies on permanent permissions (cookies, etc)

**Comment by earthlng** [comment]
* 2450c: set size of warning quota for offline cache (default 51200)
I think this is irrelevant due to `/* 1004: disable offline cache ***/` and therefore inactive is fine

**Comment by Thorin-Oakenpants** [comment]
1005 - is not recommended because it's a bit of a mystery and (i'm sure) its related to how much ram you have. `-1` is default and that's auto. Tinkering with it would probably lead to stability/performance issues.

Some of these are included for completeness. The two cookie ones we could change. Why keep cookies for 90 days? We could change that that to 10 days - but leave the prefs inactive

**Comment by earthlng** [comment]
> Tinkering with it would probably lead to stability/performance issues.

Only if you set it too high. Setting it to 0 as we have in the template should have no such impacts.
It doesn't need a warning IMO. Everything is perfectly explained in the description. If someone wants to tweak it they can look at the linked page and decide what they want.

> Why keep cookies for 90 days? We could change that that to 10 days

sure, but why keep them for 10 days? Either you want certain cookies, most likely until they expire, or you don't care about cookies and can delete them on shutdown. Both items let the user know what the default values are in FF and that's good enough IMO. No need to change anything

**Comment by earthlng** [comment]
and I also think we shouldn't add WARNING tags on inactive prefs. We should try to keep those to a minimum IMO

**Comment by Thorin-Oakenpants** [comment]
I get that, these cookie items are included for completeness. So why don't we at least harden them, albeit inactive

**Comment by Thorin-Oakenpants** [comment]
OK, I'll use [NOTE] instead of warning .. good thinking batman

**Comment by earthlng** [comment]
I don't store cookies across sessions but you do, so you can better decide what you would want those prefs to be should someone choose to activate them (in combination with the other cookie prefs).
2 + 30, 3 + 30, 3 + 10 - do whatever you think is best - IDC :)

**Comment by Thorin-Oakenpants** [comment]
I get that people like you and I etc are ok, but we have 10 inactive prefs here which are not clear cut for people to understand. Just trying to be consistent. 3 of the 4 orange ones we can include a [NOTE]. Some of the blue ones we can enforce or change the value and/or include a note.

Or we could just drop the whole thing :)

**Comment by Thorin-Oakenpants** [comment]
> Or we could just drop the whole thing :)

:+1: = drop the whole thing
:-1: = drop the whole thing

First one to one wins .. start your engines

**Comment by earthlng** [comment]
I like that the inactive ones with "set..." default to the same as FFs default. If someone wants to enforce the defaults they can just active those, or tweak them if they prefer that.
We can activate 1032. For 1031 the title is pretty self-explanatory. The other orange ones we just discussed and the remaining blue ones we can ignore IMO.

**Comment by Thorin-Oakenpants** [comment]
TBH .. considering how many numbered items we have, to only have 11 that "fail" this so called "test" is pretty telling. We're really scraping the barrel here. They were actually fine as they were. We're down to 8 right now, and if I add the info to 1003, 1005 and 1006, the last 5 are fine IMO as well. It was a good exercise, but it's getting a bit pedantic

**Comment by Thorin-Oakenpants** [comment]
Ahh .. that's interesting. PB Mode writes permanent site permission changes - i.e I made a site permission change for a site in PB Mode, closed all PB mode windows, loaded the same site in a normal window (which is how I started the browser) and the permission change has stuck. I thought PB Mode permissions were session only, which maybe they *were* - which would make life extremely hard for those starting in PB Mode. Just reading my way thru https://bugzilla.mozilla.org/show_bug.cgi?id=967812.

Wow .. this is really horrible pref IMO. I understand the reason TBB doesn't want to write anything to disk, as it would create at least a reference to having visited a site, but it really makes exceptions to default permissions a PITA
